### DISCUSSION TITLE
Parse `queryParams` to a non-primitive type in RouteArgs

### DISCUSSION THREAD

**Discussion by MartinAKovachev** [Milad-Akarie/auto_route_library#1547] [General]
Hi @Milad-Akarie,
We have a page which receives inside it's constructor a `Map<String, List<String>>` and added `@QueryParam` annotation to it.
When running the application in web and navigate to this specific page, the url bar contains the query parameters like this:
`http://test.com/#/search?query=Test+query&filters={filter1:+[a,+b],+filter2:+[c,+d]}`
After reloading the page in the browser there is an error appearing that says:

> Expected a value of type `Map<String, List<String>>?`, but got one of type `String`

After checking the generated auto route file it appeared that the queryParams is not parsed correctly:

```dart
SearchRouteArgs(
                query: queryParams.optString('query'),
                filters: queryParams.get('filters'),
              ));
``` 
Is there anything we can do to fix that issue and is it possible to pass custom object to `@PathParams` and `@QueryParams` except the primitive types?

**Comment by Milad-Akarie** [comment]
@MartinAKovachev is the url above parsable by Uri.parse?

**Reply by MartinAKovachev** [reply]
@Milad-Akarie I think it doesn't but does it matter in this case? The url is generated by AutoRoute when navigating to a certain page, which expects an object (or non-primitive type) as constructor parameter and is using @QueryParam/@PathParam. It is not possible to refresh the browser page.
